ocaml/typing/typemod.ml

```diff
@@ -3634,6 +3649,17 @@ let report_error ~loc _env = function
       Location.errorf ~loc
         "@[Top-level module bindings must have layout value, but@ \
          %s has layout@ %a.@]" id Sort.format sort
+  | Cannot_implement_parameter path ->
+      Location.errorf ~loc
+        "@[Interface %s@ found for this unit is flagged as a parameter.@ \
+         It cannot be implemented directly. Use -as-argument-for instead.@]"
```

Comment:
I don't think the suggestion is helpful here. I think making an argument for the parameter is unlikely to be what the user was trying to do.

Reply:
Changed the suggestion to "Recompile it without `-as-parameter`."